libs/tokens/src/state/tokens/allTokensAtom.ts

```diff
@@ -0,0 +1,94 @@
+import { atom } from 'jotai'
+import { environmentAtom } from '../environmentAtom'
+import { TokensMap } from '../../types'
+import { NATIVE_CURRENCY_BUY_TOKEN, TokenWithLogo } from '@cowprotocol/common-const'
+import { tokenMapToList } from '../../utils/tokenMapToList'
+import { userAddedTokensAtom } from './userAddedTokensAtom'
+import { favouriteTokensAtom } from './favouriteTokensAtom'
+import { listsEnabledStateAtom, listsStatesListAtom } from '../tokenLists/tokenListsStateAtom'
+
+export interface TokensByAddress {
+  [address: string]: TokenWithLogo
+}
+
+export interface TokensBySymbol {
+  [address: string]: TokenWithLogo[]
+}
+
+export interface TokensState {
+  activeTokens: TokensMap
+  inactiveTokens: TokensMap
+}
+
+export const tokensStateAtom = atom<TokensState>((get) => {
+  const { chainId } = get(environmentAtom)
+  const listsStatesList = get(listsStatesListAtom)
+  const listsEnabledState = get(listsEnabledStateAtom)
+
+  return listsStatesList.reduce<TokensState>(
+    (acc, list) => {
+      const isListEnabled = listsEnabledState[list.id]
+
+      list.list.tokens.forEach((token) => {
+        if (token.chainId !== chainId) return
+
+        const tokenAddress = token.address.toLowerCase()
+
+        if (isListEnabled) {
+          acc.activeTokens[tokenAddress] = token
+        } else {
+          acc.inactiveTokens[tokenAddress] = token
+        }
+      })
+
+      return acc
+    },
+    { activeTokens: {}, inactiveTokens: {} }
+  )
+})
+
+export const activeTokensAtom = atom<TokenWithLogo[]>((get) => {
+  const { chainId } = get(environmentAtom)
+  const userAddedTokens = get(userAddedTokensAtom)
+  const favouriteTokensState = get(favouriteTokensAtom)
+
+  const tokensMap = get(tokensStateAtom)
+  const nativeToken = NATIVE_CURRENCY_BUY_TOKEN[chainId]
+
+  const tokens = tokenMapToList({
```

Comment:
maybe `tokenMapToListWithLogo`? to reflect you are also adding that?

Reply:
Renamed: https://github.com/cowprotocol/cowswap/pull/3201/commits/e1bae586a70dbf1f0fe66bd8fd84a6207e0bd44c